File: src/main/java/fi/aalto/cs/apluscourses/ui/BannerPanel.java
package fi.aalto.cs.apluscourses.ui;

import com.intellij.ui.JBColor;
import com.intellij.ui.components.panels.OpaquePanel;
import com.intellij.util.ui.JBUI;
import java.awt.BorderLayout;
import java.awt.Color;
import java.awt.Dimension;
import javax.swing.BorderFactory;
import javax.swing.JLabel;
import javax.swing.JPanel;
import org.jetbrains.annotations.NotNull;

public class BannerPanel extends JPanel {
  @NotNull
  private final JLabel bannerText = new JLabel();

  @NotNull
  private final JPanel containerPanel;

  /**
   * Constructor for the banner.
   */
  public BannerPanel() {
    super(new BorderLayout());

    containerPanel = new OpaquePanel(new BorderLayout());
    containerPanel.add(BorderLayout.CENTER, bannerText);
    containerPanel.setBorder(JBUI.Borders.empty(5, 0, 5, 5));
    containerPanel.setMinimumSize(new Dimension(0, 0));

    add(BorderLayout.CENTER, containerPanel);

    setBorder(BorderFactory.createCompoundBorder(
        BorderFactory.createMatteBorder(0, 0, 1, 0, JBColor.border()),
        BorderFactory.createEmptyBorder(0, 10, 0, 10))
    );
  }

  public void setText(String text) {
    bannerText.setText(text);
  }

  public void setBannerColor(@NotNull Color color) {
    containerPanel.setBackground(color);
    setBackground(color);
  }
}

File: src/main/kotlin/fi/aalto/cs/apluscourses/ui/BannerPanel.kt
package fi.aalto.cs.apluscourses.ui

import com.intellij.ui.components.JBLabel
import com.intellij.util.ui.JBUI
import com.intellij.util.ui.JBUI.CurrentTheme.NotificationError
import java.awt.BorderLayout
import java.awt.Color
import javax.swing.BorderFactory
import javax.swing.JPanel

class BannerPanel(text: String, bannerType: BannerType) : JPanel(BorderLayout()) {
    private val errorForeground: Color = NotificationError.foregroundColor()
    private val errorBackground: Color = NotificationError.backgroundColor()
    private val errorBorderColor: Color = NotificationError.borderColor()

    /**
     * Constructor for the banner.
     */
    init {
        val bannerText = JBLabel(text)
        bannerText.border = JBUI.Borders.empty(6)
        add(bannerText)
        when (bannerType) {
            BannerType.ERROR -> {
                bannerText.foreground = errorForeground
                background = errorBackground
                border = BorderFactory.createMatteBorder(0, 0, 1, 0, errorBorderColor)
            }
        }
    }

    override fun getWidth(): Int {
        return parent.width
    }

    enum class BannerType {
        ERROR,
//        WARNING,
//        INFO
    }

}
